Lib/_pyrepl/windows_console.py

```diff
@@ -521,7 +535,23 @@ def forgetinput(self) -> None:
     def getpending(self) -> Event:
         """Return the characters that have been typed but not yet
         processed."""
-        return Event("key", "", b"")
+        e = Event("key", "", b"")
+
+        while not self.event_queue.empty():
+            e2 = self.event_queue.get()
+            if e2:
+                e.data += e2.data
+
+        recs, rec_count = self._read_input_bulk(False, 1024)
+        for i in range(rec_count):
+            rec = recs[i]
+            if rec and rec.EventType == KEY_EVENT:
+                key_event = rec.Event.KeyEvent
+                ch = key_event.uChar.UnicodeChar
+                if ch == "\r":
```

Comment:
We later do:

```
data = str(raw, self.encoding, "replace")
```

this can cause us to get  `\r\n\n,` which incorrectly introduces blank lines on Windows, no? Maybe I am reading this wrong